onnx/shape_inference/implementation.h

```diff
@@ -325,15 +325,34 @@ struct DataPropagationContextImpl : public DataPropagationContext {
     }
     return &allOutputTypes_[index];
   }
-
-  const TensorProto* getInputData(size_t index) const override {
+  const TensorShapeProto* getInputData(size_t index) const override {
     if (index >= allInputData_.size()) {
       ONNX_THROW("Input " + ONNX_NAMESPACE::to_string(index) + " is out of bounds.");
     }
-    return allInputData_[index];
+    // Gets it from initializer if it exists
+    const auto* input_data = allInputData_[index];
+    if (input_data != nullptr) {
+      std::vector<int64_t> input_vals;
+      if (input_data->data_type() == TensorProto_DataType_INT64) {
+        input_vals = ParseData<int64_t>(input_data);
+      } else if (input_data->data_type() == TensorProto_DataType_INT32) {
+        std::vector<int32_t> input_vals = ParseData<int32_t>(input_data);
```

Comment:
This is a different variable from the outer-scope variable. This won't work since it won't be used in the loop below in line 342. I think you may have to duplicate the loop lines 342-344 separately for the then branch and else branch.

Reply:
Updated. Thanks for the catch